--- maze.c	2023-11-24 16:39:14.960007393 +0300
+++ maze-2.c	2023-11-24 16:51:40.593056628 +0300
@@ -56,6 +56,19 @@
 	char pass = '.', wall = '#';
 	char *maze;
 
+	if(argc < 3) {
+		printf("Usage: %s <pass,wall> <size>\n", argv[0]);
+		return 1;
+	}
+
+	pass = argv[1][0];
+	wall = argv[1][1];
+	size = atoi(argv[2]);
+	if (size < 1) {
+		printf("Wrong maze size\n");
+		return 1;
+	}
+
 	srand(time(0));
 
 	width = size * 2 + 1;
